src/output/plugins/OSXOutputPlugin.cxx

```diff
@@ -717,7 +709,28 @@ osx_render(void *vdata,
 {
 	OSXOutput *od = (OSXOutput *) vdata;
 
-	int count = in_number_frames * od->asbd.mBytesPerFrame;
+	if (od->cancel) {
+		od->ring_buffer->reset();
+		od->cancel = false;
+	}
+
+	// Wait until data is available.
+	while (!od->ring_buffer->read_available());
```

Comment:
This is a busy loop, churning CPU and thus slowing down the decoder/player threads, leading to further delay. Is this really a good idea?

Reply:
The code will be removed in a follow up PR, as you want to split the silence generation code from the bigger PR. When silence generation is added this will become an if rather than while. 

Also this only busy waits the moment Cancel happens and running for a couple of milliseconds at most. So the impact is very minimal.

Reply:
I disagree, it's not "minimal". If `osx_render()` runs with real-time priority, this will completely block one CPU core and will not give anybody else a share.
I don't even know why this loop is part of the commit. It has nothing to do with cancellation. The loop didn't exist previously, and the "real" changes in this commit don't add a need for it.
Just like silence generation didn't belong in this commit - but if you feel the silence generator is necessary to avoid this busy loop, then let's merge the silence generator commit first!

Reply:
i lost interest in keeping up with the ever changing requests.  
Sorry I don’t have that much time to change the code back and forth. 

If someone else has interest in merging the changes in I will be happy to see that happen 

Feel free to close these pull requests